rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -143,34 +218,37 @@ Player::Player(
   {
     std::lock_guard<std::mutex> lk(reader_mutex_);
     reader_ = std::move(reader);
-    // keep reader open until player is destroyed
-    reader_->open(storage_options_, {"", rmw_get_serialization_format()});
-    auto metadata = reader_->get_metadata();
-    starting_time_ = std::chrono::duration_cast<std::chrono::nanoseconds>(
-      metadata.starting_time.time_since_epoch()).count();
-    // If a non-default (positive) starting time offset is provided in PlayOptions,
-    // then add the offset to the starting time obtained from reader metadata
-    if (play_options_.start_offset < 0) {
-      RCLCPP_WARN_STREAM(
-        get_logger(),
-        "Invalid start offset value: " <<
-          RCUTILS_NS_TO_S(static_cast<double>(play_options_.start_offset)) <<
-          ". Negative start offset ignored.");
-    } else {
-      starting_time_ += play_options_.start_offset;
-    }
-    clock_ = std::make_unique<rosbag2_cpp::TimeControllerClock>(
-      starting_time_, std::chrono::steady_clock::now,
-      std::chrono::milliseconds{100}, play_options_.start_paused);
-    set_rate(play_options_.rate);
-    topic_qos_profile_overrides_ = play_options_.topic_qos_profile_overrides;
-    prepare_publishers();
-    configure_play_until_timestamp();
+    finish_constructor();
```

Comment:
Suggestion:
- Create a protected `void init` method instead of the `finish_constructor()` with the following parameters
```cpp
void init(const rosbag2_storage::StorageOptions & storage_options,
  const rosbag2_transport::PlayOptions & play_options) 
{
     storage_options_ = storage_options;
     play_options_ = play_options;
     // Content of the current finish_constructor()
}
```
Also curious about how to handle 
```
rclcpp::NodeOptions(node_options).arguments(play_options.topic_remapping_options)
```

Reply:
Ah yes, yesterday I forgot to commit the assignment of the option sets :)
For the remapping, I think this overlaps with classic ros2 remapping (from ros2 run CLI, or from launchfile). We shall consider to enforce remapping in the classic way, and leave this empty; it is still useful and valid instead for "ros2 bag play" verb.

Reply:
Maybe add a check that if `play.topic_remapping_options` is not empty then log a warning that it is not expected and topic remapping shall be used as classic CLI/launch remapping.